internal/server/grpc/handlers/secretHandler.go

```diff
@@ -0,0 +1,141 @@
+// Package handlers содержит обработчики gRPC-запросов, связанных с управлением секретами пользователей.
+package handlers
+
+import (
+	"beliaev-aa/GophKeeper/internal/server/service"
+	"beliaev-aa/GophKeeper/pkg/consts"
+	"beliaev-aa/GophKeeper/pkg/converter"
+	"beliaev-aa/GophKeeper/pkg/proto"
+	"context"
+	"errors"
+	"fmt"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"strconv"
+)
+
+// SecretHandler реализует серверные функции для управления секретами пользователей.
+type SecretHandler struct {
+	proto.UnimplementedSecretsServer
+	logger              *zap.Logger
+	secretService       *service.SecretService
+	notificationHandler *NotificationHandler
+}
+
+// NewSecretHandler создаёт новый экземпляр сервера для управления секретами.
+// Возвращает инициализированный экземпляр SecretHandler.
+func NewSecretHandler(logger *zap.Logger, secretService *service.SecretService) *SecretHandler {
+	return &SecretHandler{
+		logger:              logger,
+		notificationHandler: NewNotificationHandler(logger),
+		secretService:       secretService,
+	}
+}
+
+// SaveUserSecret сохраняет или обновляет секрет пользователя.
+// Возвращает пустой ответ или ошибку, если операция не удалась.
+func (s *SecretHandler) SaveUserSecret(ctx context.Context, in *proto.SaveUserSecretRequest) (*emptypb.Empty, error) {
+	userID, err := extractUserID(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	secret := converter.ProtoToSecret(in.Secret)
+	secret.UserID = int(userID)
+	if secret.ID > 0 {
+		_, err = s.secretService.UpdateSecret(ctx, secret)
+	} else {
+		_, err = s.secretService.CreateSecret(ctx, secret)
+	}
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	clientID, err := extractClientID(ctx)
+	if err == nil {
+		isUpdated := secret.ID > 0
+		err = s.notificationHandler.notifyClients(userID, clientID, secret.ID, isUpdated)
+		if err != nil {
+			s.logger.Error("failed to notify clients", zap.Error(err))
+		}
+	}
+	return &emptypb.Empty{}, nil
+}
+
+// GetUserSecret извлекает конкретный секрет пользователя.
+// Возвращает секрет или ошибку, если секрет не найден или запрос не может быть выполнен.
+func (s *SecretHandler) GetUserSecret(ctx context.Context, in *proto.GetUserSecretRequest) (*proto.GetUserSecretResponse, error) {
+	userID, err := extractUserID(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	secret, err := s.secretService.GetSecret(ctx, in.Id, userID)
+	if errors.Is(err, fmt.Errorf("secret not found (id=%d)", in.Id)) {
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
```

Comment:
Стоит проверить наличие ошибки и потом уже ее тип.
```suggestion
	if err != nil {
	        if errors.Is(err, fmt.Errorf("secret not found (id=%d)", in.Id)) {
		        return nil, status.Error(codes.NotFound, err.Error())
	        }
		return nil, status.Error(codes.Internal, err.Error())
	}
```

Reply:
Исправил